### ISSUE TITLE
Instructor extending individual deadlines: table doesn't sort by team

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#12596]
v8.29, production

Context: an instructor is in the page that is for extending deadlines.

![image](https://github.com/TEAMMATES/teammates/assets/1673303/14850ef8-be6a-471a-8ccc-72327d4df495)

Problem: when trying to sort by the `Team` column, the row order stays the same i.e., nothing happens when the up/down icons are clicked. But no such issue with the other columns of the table.

![image](https://github.com/TEAMMATES/teammates/assets/1673303/ad6a7e11-5570-40cb-9e35-92432f1478b6)

**Comment by MaharshiPedu** [comment]
Hi @damithc, I could not reproduce this issue. The table does get sorted according to `Team` column.

**Comment by damithc** [comment]
@MaharshiPedu the case I looked at had a few ongoing extensions already. Perhaps that is the trigger for this bug. Can you try that too?

**Comment by MaharshiPedu** [comment]
@damithc  I tried what you suggested. In spite of giving multiple extensions, the order of sorting of `Team` column did change by using the arrows beside the column heading.

However, I found that the sorting order is wrong. `Team 2` appears before `Team 3` if sorted in descending order. Also, in ascending order, the order is incorrect. One `Team 3` row appears between other `Team 2` rows. Have a look at the gif below to get a clear understanding:

![Peek 2023-09-30 17-58](https://github.com/TEAMMATES/teammates/assets/84611938/c4cb226e-1d8c-4b27-b06a-da84887934ac)

**Comment by MaharshiPedu** [comment]
@damithc I tried what you suggested. In spite of giving multiple extensions, the order of sorting of Team column did change by using the arrows beside the column heading.

However, I found that the sorting order is wrong. In ascending order, the order is incorrect. One Team 3 row appears between other Team 2 rows. Have a look at the gif below to get a clear understanding:

![Peek 2023-09-30 18-12](https://github.com/TEAMMATES/teammates/assets/84611938/75f6020e-cbe6-4de9-8351-aaa3a90e2bb6)

**Comment by damithc** [comment]
@MaharshiPedu Also note that the bug I reported is in the production version which may be a number of commits behind the master branch. So, the behavior you see may be different from the one I reported. 

@cedricongjh is this behavior potentially affected by recent PRs that we merged? I remember there was a major PR that was related to table sorting.

**Comment by cedricongjh** [comment]
yes there was this PR: https://github.com/TEAMMATES/teammates/pull/12556 that was merged in since the last release, hence the difference in behaviour, it could be worth looking into why the team sort (and the other sorts as well) has an issue when sorted by ascending as @MaharshiPedu pointed out

**Comment by rameshmoorjani** [comment]
Hello @damithc - I am beginner in open source contribution. Can I participate and work on this issue?

**Comment by cedricongjh** [comment]
hey @rameshmoorjani, feel free to look into this and open a PR

**Comment by wanqiqiu** [comment]
hi, @cedricongjh  can I work on this issue to have a try? Thank you!
(uid: u7503408 for reference.)

**Comment by cedricongjh** [comment]
hi @Ruimeng-Ye and @wanqiqiu, since @rameshmoorjani has expressed interest in this issue, let's give him time to work on it, you're welcome to look into other issues to contribute to!

@rameshmoorjani do let us know if you're still working on it, thank you!

**Comment by rameshmoorjani** [comment]
@cedricongjh - yes I am working on it, having some issue with setup in local due to ng-in-viewport and Agular version mismatch. I will update asap.

**Comment by emanuelconte** [comment]
Hi @damithc . Can I work on this issue ?

I'm new to open source projects and would like to give this a try.

**Comment by azharmd1** [comment]
I looked into the code base.  I would like to work on this issue. Please assign me!

**Comment by WesleyJammer** [comment]
hey @damithc, i see this project is still open, so is it possible for me to work on this? i am getting into opensource trough my study and this looks interessting.

**Comment by VeldiBharath** [comment]
is this issue still open? I would like to contribute !

**Comment by dishenggg** [comment]
Hi @VeldiBharath, yes feel free to submit a PR!

**Comment by Hsuhx** [comment]
Hi @damithc, can I have a try with this issue? Thank you.

**Comment by serene-breeze** [comment]
Interested in this one and taking a look at it,

**Comment by Gambit-02** [comment]
Hi, I'd like to take on this issue as part of a software engineering course project at the University of Michigan (EECS 481). I noticed there is a cross-fork PR linked from another contributor, but it does not appear to have been submitted upstream, so I'm assuming this is still open for contribution. If that's incorrect, please let me know and I'll pick a different issue. Otherwise, I plan to investigate the sort comparator for the deadline extension table and open a pull request within the next two to three weeks. Thanks!
